### PR TITLE
D8CORE-322 Use domain redirect and config pages to allow custom domains

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Use config_pages module to create a page for users to set values for the site
- set up domain 301 redirect to redirect all traffic to a canonical url.

# Need Review By (Date)
- 9/25

# Urgency
- medium

# Steps to Test
1. `composer require su-sws/stanford_profile:dev-D8CORE-322-domain-redirect`
1. `drush cim -y`
1. navigate to `/admin/config/system/basic-site-settings`
1. enter a url. anything URL that is different from your current will work & save
1. view the site as anonymous
1. validate it redirects you off to the domain you configured

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by jbickar** [comment]
I'm not sure how I feel about a [contrib module that not only does not have a stable release, but does not have a D8 release at all](https://www.drupal.org/project/domain_301_redirect).

**Comment by pookmish** [comment]
I've reviewed the module very closely and it's been in use on humsci for a long time. its the only real option in the contrib space. The only other option i can find is for us to build an identical solution.

**Comment by jbickar** [comment]
Maybe you should offer to become a maintainer? 😂

**Comment by pookmish** [comment]
hard pass

**Comment by jbickar** [comment]
Can you review https://www.drupal.org/project/domain_301_redirect/issues/2775825 to get it moving?

**Comment by sherakama** [comment]
Modules on a development version with no release have a few extra requirements before adding.

1. Code review by two developers
2. Locking to a specific version
3. No patches just forks

* If the code review passes muster and there are no eyebrow-raising issues the module can be added but locked to a hash
* If the code review fails for security reasons but can be fixed, we adopt a fork of the module 
* If the dev code is good be requires one or more patches we fork the module and apply the patch to our fork.
* By forking the module we take ownership and then subject the module to our development standards and the definition of done.

**Comment by pookmish** [comment]
That all sounds great and dandy. so basically it needs another pair of :eyes: on the code. the composer.json in this PR has it locked to a commit and its not needing any patches. H&S is using a patch i created https://www.drupal.org/project/domain_301_redirect/issues/3010370 which basically redirects for all users including admin role. the module will redirect for anonymous and non-admin role OOTB though. so it'll get what we need i think

On another note. i'm curious to get thoughts on the use of config_pages.

**Comment by pookmish** [comment]
ask and you shall receive https://www.drupal.org/project/domain_301_redirect/releases/8.x-1.0-alpha0

**Comment by jbickar** [comment]
>>> What is the delay in getting the above fixes added. We've been using the dev version of this module for many months and it's proved to withstand over a dozen sites at this time. Can we move this along and get at least an alpha or beta release out.

"How to Win Friends and Influence People", by @pookmish.

File: composer.json [hunk 1]
```diff
@@ -59,7 +59,9 @@
         "su-sws/stanford_media": "dev-8.x-1.x",
         "su-sws/stanford_paragraph_types": "dev-8.x-2.x",
         "su-sws/stanford_ssp": "dev-8.x-1.x",
-        "su-sws/stanford_text_editor": "dev-8.x-1.x"
+        "su-sws/stanford_text_editor": "dev-8.x-1.x",
+        "drupal/domain_301_redirect": "dev-1.x#fd66e82fefff76fddc5968286108056cd17010a4",
```
Comment (sherakama) [inline composer.json]: This can go to the alpha release now.

**Comment by sherakama** [comment]
I'd like @boznik to chime in on the config pages approach.

**Comment by pookmish** [comment]
I would like to explore the ability to make the overrides more dynamic. Like a connecting page/config that says this config_pages page & this field value override this other system config value... Im not sure how i forsee it being wired up in the UI yet

**Comment by boznik** [comment]
don't forget to clear the cache for it to take effect.

**Pull request merged**